library/Freckle/App/Database.hs

```diff
@@ -65,12 +71,68 @@ makePostgresPool = do
   makePostgresPoolWith conf
 
 runDB
-  :: (HasSqlPool app, MonadUnliftIO m, MonadReader app m)
+  :: ( HasSqlPool app
+     , Stats.HasStatsClient app
+     , MonadHandler m
+     , MonadUnliftIO m
+     , MonadReader app m
+     )
   => SqlPersistT m a
   -> m a
 runDB action = do
+  app <- ask
   pool <- asks getSqlPool
-  runSqlPool action pool
+  mVaultData <- vaultDataFromRequest <$> waiRequest
+  maybe
+    runSqlPool
+    (runSqlPoolXRay "runDB")
+    mVaultData
+    (bracket_ (increment app) (decrement app) action)
+    pool
+  where
+    increment = runReaderT $ Stats.withGauge "db.active-connections" (withEKG (`EKG.add` 1))
+    decrement = runReaderT $ Stats.withGauge "db.active-connections" (withEKG (`EKG.subtract` 1))
+    withEKG f gauge = do
+      liftIO $ f gauge >> EKG.read gauge
```

Comment:
I think it's important not to leak the `EKG` stuff. It's also a bug risk to have to type this name twice. Can something like this work?

```hs
withGauged "db.active-connections" $ do
  maybe
    runSqlPool
    (runSqlPoolXRay "runDB")
    mVaultData
    action
    pool
```

Where `withGauged` is a single encapsulating function in `Freckle.App.Stats`.

I figured I'd raise this in prototype stage, because I _assume_ you'd have to move the wrapper to outside of runner there. In other words, this won't work:

```hs
  maybe
    runSqlPool
    (runSqlPoolXRay "runDB")
    mVaultData
    (withGauged "db.active-connections" action)
    pool
```

Becase `withGauged` needs `MonadReader env m, HasStatsClient env` and `SqlPersistT m` can't supply that. And the bracket-like nature of it means a `lift` can't help either.

There may be a way to get around that, which would be ideal, because I don't know if there are any negative effects to wrapping the increment/decrement outside of, instead of within, the runner.

WDYT?

Reply:
If we only intend to use withGauged with increment/decrement then I think we could wrap the entire maybe.

So to be clear, you think EKG should be internal to Stats and not leak out? That's probably a good goal.

Reply:
> So to be clear, you think EKG should be internal to Stats and not leak out? 

Yup. It's not any kind of deal breaker, but it seems attainable to keep that an implementation detail so I think we should.